lib/matplotlib/tests/test_legend.py

```diff
@@ -1068,6 +1068,204 @@ def test_legend_labelcolor_rcparam_markerfacecolor_short():
+    leg = ax.legend(labelcolor='linecolor')
+    assert_last_legend_patch_color(h, leg, 'r', facecolor=True)
+
+    h = ax.hist(x, histtype='step', color='g', label="green step hist, green label")
+    leg = ax.legend(labelcolor='linecolor')
+    assert_last_legend_patch_color(h, leg, 'g', edgecolor=True)
+
+    h = ax.hist(x, histtype='stepfilled', color='b',
+                label="blue stepfilled hist with a blue label")
+    leg = ax.legend(labelcolor='linecolor')
+    assert_last_legend_patch_color(h, leg, 'b', facecolor=True)
+
+    # testing c, fc, and ec combinations for bar histograms
+    h = ax.hist(x, histtype='bar', color='r', ec='b',
+                label="red bar hist with blue edges and a red label")
+    leg = ax.legend(labelcolor='linecolor')
+    assert_last_legend_patch_color(h, leg, 'r', facecolor=True)
+
+    h = ax.hist(x, histtype='bar', fc='r', ec='b',
+                label="red bar hist with blue edges and a red label")
+    leg = ax.legend(labelcolor='linecolor')
+    assert_last_legend_patch_color(h, leg, 'r', facecolor=True)
+
+    h = ax.hist(x, histtype='bar', fc='none', ec='b',
+                label="unfilled blue bar hist with a blue label")
+    leg = ax.legend(labelcolor='linecolor')
+    assert_last_legend_patch_color(h, leg, 'b', edgecolor=True)
+
+    # testing c, and ec combinations for step histograms
+    h = ax.hist(x, histtype='step', color='r', ec='b',
+                label="blue step hist with a blue label")
+    leg = ax.legend(labelcolor='linecolor')
+    assert_last_legend_patch_color(h, leg, 'b', edgecolor=True)
+
+    h = ax.hist(x, histtype='step', ec='b',
+                label="blue step hist with a blue label")
+    leg = ax.legend(labelcolor='linecolor')
+    assert_last_legend_patch_color(h, leg, 'b', edgecolor=True)
+
+    # testing c, fc, and ec combinations for stepfilled histograms
+    h = ax.hist(x, histtype='stepfilled', color='r', ec='b',
+                label="red stepfilled hist, blue edges, red label")
+    leg = ax.legend(labelcolor='linecolor')
+    assert_last_legend_patch_color(h, leg, 'r', facecolor=True)
+
+    h = ax.hist(x, histtype='stepfilled', fc='r', ec='b',
+                label="red stepfilled hist, blue edges, red label")
+    leg = ax.legend(labelcolor='linecolor')
+    assert_last_legend_patch_color(h, leg, 'r', facecolor=True)
+
+    h = ax.hist(x, histtype='stepfilled', fc='none', ec='b',
+                label="unfilled blue stepfilled hist, blue label")
+    leg = ax.legend(labelcolor='linecolor')
+    assert_last_legend_patch_color(h, leg, 'b', edgecolor=True)
+
+    h = ax.hist(x, histtype='stepfilled', fc='r', ec='none',
+                label="edgeless red stepfilled hist with a red label")
+    leg = ax.legend(labelcolor='linecolor')
+    assert_last_legend_patch_color(h, leg, 'r', facecolor=True)
+    plt.close('all')
+
+
+def assert_last_legend_linemarker_color(plot, leg, expected_color,
```

Comment:
```suggestion
def assert_last_legend_linemarker_color(line, leg, expected_color,
```
Let's call this by the created artist. Also I suggest, to use tuple-unpacking on the plot call sites (which is a standard pattern we recommend, see https://matplotlib.org/devdocs/tutorials/pyplot.html#controlling-line-properties)
```
l, = ax.plot(x, c='r', label="red line with a red label")
```
so that the assert can accept a single Line2D instance and you dont have to do `plot_marker = plot[0]`.

Reply:
done in https://github.com/matplotlib/matplotlib/pull/30328/commits/7d2f76bba882084e8bde77eba7f67659248c2247